### PR TITLE
Feat: Align `go-redis` `TimeseriesCmdable`

### PR DESCRIPTION
**Pull request by unknowntpo**
This PR implemented adapter for `go-redis` `TimeseriesCmdable`.

### CODE REVIEW

File: rueidiscompat/adapter.go [hunk 1]
```diff
@@ -3513,6 +3546,792 @@ func (c *Compat) TDigestTrimmedMean(ctx context.Context, key string, lowCutQuant
...
+		}
+		if options.BucketTimestamp != nil {
+			cmds.TsMrangeAggregationBucketduration(_cmd).Buckettimestamp(str(options.BucketTimestamp))
+		}
+		if options.Empty {
+			cmds.TsMrangeAggregationBuckettimestamp(_cmd).Empty()
+		}
+		cmds.TsMrangeTotimestamp(_cmd).Filter(filterExpr...)
+		if options.GroupByLabel != nil {
+			// FIXME: Wrong API definition: REDUCE
```
Comment (unknowntpo) [inline rueidiscompat/adapter.go:4196]: Note that our builder API is wrong.

File: rueidiscompat/adapter.go [hunk 2]
```diff
@@ -3513,6 +3546,792 @@ func (c *Compat) TDigestTrimmedMean(ctx context.Context, key string, lowCutQuant
...
+		}
+		if options.BucketTimestamp != nil {
+			cmds.TsMrevrangeAggregationBucketduration(_cmd).Buckettimestamp(str(options.BucketTimestamp))
+		}
+		if options.Empty {
+			cmds.TsMrevrangeAggregationBuckettimestamp(_cmd).Empty()
+		}
+		cmds.TsMrevrangeTotimestamp(_cmd).Filter(filterExpr...)
+		if options.GroupByLabel != nil {
+			// FIXME: Wrong API definition: REDUCE
```
Comment (unknowntpo) [inline rueidiscompat/adapter.go:4295]: Note that our builder API is wrong.

File: rueidiscompat/command.go [hunk 3]
```diff
@@ -3170,3 +3170,305 @@ type TDigestMergeOptions struct {
...
+func newMapStringInterfaceCmd(res rueidis.RedisResult) *MapStringInterfaceCmd {
+	cmd := &MapStringInterfaceCmd{}
+	m, err := res.AsMap()
+	if err != nil {
+		cmd.err = err
+		return cmd
+	}
+	strIntMap := make(map[string]any, len(m))
+	for k, ele := range m {
+		var v any = rueidis.Nil
```
Comment (unknowntpo) [inline rueidiscompat/command.go:3311]: This `rueidis.Nil` is added to make tests pass, should we use `rueidis.Nil` or let it be `nil` if `ele.IsNil()` is true ?

Comment (rueian) [reply rueidiscompat/command.go:3311]: I think `nil` is more reasonable.

File: message.go [hunk 4]
```diff
@@ -1130,6 +1130,28 @@ func (m *RedisMessage) AsGeosearch() ([]GeoLocation, error) {
 	return geoLocations, nil
 }
 
+// AsAnyAnyMap converts RedisMessage to map[any]any.
+// we don't use (*RedisMessage).ToAny() on map type because it will convert ele
+// to map[string]any
+func (m *RedisMessage) AsAnyAnyMap() (map[any]any, error) {
```
Comment (rueian) [inline message.go:1136]: `map[any]any` is too hard to use. I don't think we should have this function.

Comment (unknowntpo) [reply message.go]: Okay, I'll remove it.

**Pull request merged**